middleware/proxy/upstream.go

```diff
@@ -150,6 +153,14 @@ func RegisterPolicy(name string, policy func() Policy) {
 	supportedPolicies[name] = policy
 }
 
+// AddProxyHeader adds a proxy header.
+func addProxyHeader(header, value string) {
+	if proxyHeaders == nil {
+		proxyHeaders = make(map[string][]string)
```

Comment:
I see your point - let's do it for now, though, since I foresee a refactoring of this package later, which should address this and a few other things.